client/src/components/VideosList.jsx

```diff
@@ -0,0 +1,105 @@
+import React, { useEffect, useState } from "react";
+import Video from "./Video";
+import AddVideo from "./AddVideo";
+import "../index.css";
+import Search from "./Search";
+
+function VideosList() {
+  const [videos, setVideos] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [searchResult, setSearchResult] = useState([]);
+
+  useEffect(() => {
+    fetch("https://node-server-full-stack.onrender.com/videos")
+      .then((res) => res.json())
+      .then((data) => {
+        console.log(data);
+        setVideos(data);
+        setLoading(false);
+      })
+      .catch((error) => {
+        console.error("Error fetching data:", error);
+      });
+  }, []);
+
+  const newVideo = (newInput) => {
+    newInput.id = videos.length + 1;
+    newInput.rating = 0;
+    fetch("https://node-server-full-stack.onrender.com/videos", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/JSON",
+      },
+      body: JSON.stringify(newInput),
+    })
+      .then((res) => {
+        fetchVideos();
+        return res.json();
+      })
+      .then((data) => {
+        if (data && data.message === "Video stored successfully") {
```

Comment:
Now you're calling `setVideos` down here as well - I think you need to think very carefully about what happens when after you've POSTed a new video to your database